cmd/aterm/config/config.go

```diff
@@ -1,156 +1,228 @@
-// Copyright 2020, Verizon Media
-// Licensed under the terms of the MIT. See LICENSE file in project root for terms.
-
 package config
 
 import (
-	"encoding/base64"
-	"flag"
+	"fmt"
 	"io"
 	"io/ioutil"
+	"net/url"
 	"os"
 	"path/filepath"
 
+	"github.com/adrg/xdg"
 	"github.com/kelseyhightower/envconfig"
-	"github.com/pkg/errors"
+	"github.com/theparanoids/aterm/errors"
+	"github.com/theparanoids/aterm/network"
 	"gopkg.in/yaml.v2"
+
+	"github.com/hashicorp/go-multierror"
 )
 
-var configFilePath = filepath.Join(os.Getenv("HOME"), ".config", "ashirt", "term-recorder.yaml")
+func configHome() string {
+	return filepath.Join(xdg.ConfigHome, "ashirt")
```

Comment:
The library we're using here is wrong for macos and is using $HOME/Library/Preferences rather than $HOME/Library/Application Support. From my understanding the preferences directory is used specifically for serialized plists that store application information. The ASHIRT client is also using the Preferences directory which is something that I wanted to change before we tag a 1.0. If we want to stick with unix style paths for this (eg. $HOME/.config/ashirt) that's also fine.

Reply:
I think we can use this XDG library instead: https://github.com/OpenPeeDeeP/xdg
I thought I had read that MacOS doesn't strictly follow xdg guidelines, so that might be why we're seeing the discrepancy. I believe I chose this one to match where ASHIRT config goes. I believe Qt is in control of that aspect more closely, but I'll double check that.

There's also a matter of migrating over older configs, if that's valuable. The newer config is not quite one-to-one, but the core configurations, obviously, are present. If we _do_ do that, then I think it'll just be a command line flag like `aterm -migrate` and have it look for the `$HOME/.config/ashirt/whatever.yaml` file. Maybe something more specific than migrate, in case we want to reserve that for later use.